z3/src/ast.rs

```diff
@@ -196,6 +196,25 @@ pub trait Ast<'ctx>: Sized + fmt::Debug + Into<Dynamic<'ctx>> {
         })
     }
 
+
+    /// Compare this `Ast` with another `Ast`, and get a Result.  Errors if the sort does not
+    /// match for the two values.
+    fn _safe_eq(&self, other: &Self) -> Result<Bool<'ctx>, std::string::String> {
```

Comment:
now that there's a safe method could you make `_eq` call it? That way z3 can unwrap with the error message instead of just the opaque assertion.